### ISSUE TITLE
API: No timeout for Wasabi operations

### ISSUE THREAD

**Issue by benoit74** [openzim/zimfarm#863]
Operations performed on Wasabi (typically updating the S3 policy when a worker IP changes) have no timeout.

This means that DB locks arise when Wasabi is not responding because all subsequent worker requests are trying to do the same thing which is blocked

We should:
- implement a timeout for Wasabi operations (maybe just monitor the operation as a whole) with a nice log message
- add a Grafana alert for these timeouts
- probably commit the DB changes before calling Wasabi, so that we do not mind if this works or not, or at least add a try/catch around Wasabi operations

On the long term, just like CMS operations, this should be moved to an asynchronous job

**Comment by elfkuzco** [comment]
Implementing a timeout for requests is an easy fix but I am not sure on how to go about the Grafana integration given I don't much how it fits in our system.

The last approach of committing before calling Wasabi is already in place but if we move to an asynchronous job by using FastAPI's `BackgroundTask`, wouldn't we be suffering some race conditions? Plus, currently, we return a server error but if we offload to `BackgroundTask`, we wouldn't have a way of raising the error. Or is that expendable?

**Comment by benoit74** [comment]
Indeed, the part regarding DB commit and try/catch block is already implemented since almost one year and it looks like it works fine. Let's keep it as-is for now, I agree that moving to an asynchronous job might prove to cause more problems than it would solve. I strongly suspect we will revisit it at some point, but there is no immediate need and no clear solution or benefit

Regarding timeout, beware that the timeouts we are dealing with are regarding s3 operations, not regular requests handled by the code. All s3 operation are going through `KiwixStorage` class implemented in [python-storagelib](https://github.com/openzim/python-storagelib), which in turn uses boto3 library. I'm pretty sure it is pretty easy to implement timeouts here as well, just a bit more complex than regular requests. I don't know which timeout is currently used and which one would be fine, but given these s3 operations are quite simple + we are supposed to have good connectivity, I'm sure we could be quite more aggressive than default values which are probably quite conservative.

Regarding Grafana monitoring, I've just added an alarm whenever we have `Pushing IP changes to Wasabi failed` in the logs: https://kiwixorg.grafana.net/alerting/grafana/deyivvb9gsy68f/view ; consider this part is done.

**Comment by elfkuzco** [comment]
Created openzim/python-storagelib#12 to allow users to specify custom configuration values like timeouts. If this is accepted, we could configure our instances to use timeouts.

**Comment by benoit74** [comment]
waiting for https://github.com/openzim/python-storagelib/issues/15 and release
